pallets/automation-time/src/lib.rs

```diff
@@ -1345,10 +1357,29 @@ pub mod pallet {
 				Self::is_valid_time(*time)?;
 			}
 
+			// Execution fee
 			let fee =
 				Self::calculate_execution_fee(&action, execution_times.len().try_into().unwrap());
-			T::FeeHandler::can_pay_fee(&who, fee.clone())
-				.map_err(|_| Error::<T>::InsufficientBalance)?;
+
+			// XCMP fee
+			let xcmp_fee: u128 = match action {
+				Action::XCMP { para_id, currency_id, encoded_call_weight, .. } =>
+					T::XcmpTransactor::get_xcm_fee(
+						u32::from(para_id),
+						currency_id,
+						encoded_call_weight.clone(),
+					)
+					.unwrap()
```

Comment:
is `unwrap()` correct here? I thought it caused a panic but could easily be wrong